author	Tony Ambardar <itugrok@yahoo.com>	Sun Apr 18 03:09:44 2021 -0700
committer	Kevin Darbyshire-Bryant <ldir@darbyshire-bryant.me.uk>	Thu Apr 22 15:11:36 2021 +0100
iproute2: avoid unneeded compiles to speed building

Skip building Makefile targets that aren't packaged: tipc, dcb, ifstat,
rtacct, lnstat, and man. Also, only compile targets needed for the current
build variant i.e. don't compile 'tc' when building an 'ip' variant and
vice versa.

These changes reduce typical build times by over 30%:
  $ make package/iproute2/clean && time make -j8 package/iproute2/compile
  (old)
  ...
  real    2m24.985s
  user    3m12.537s
  sys     0m26.677s

  (new)
  ...
  real    1m36.945s
  user    2m8.734s
  sys     0m20.046s

Signed-off-by: Tony Ambardar <itugrok@yahoo.com>
--- a/misc/Makefile
+++ b/misc/Makefile
@@ -2,13 +2,13 @@
 SSOBJ=ss.o ssfilter_check.o ssfilter.tab.o
 LNSTATOBJ=lnstat.o lnstat_util.o
 
-TARGETS=ss nstat ifstat rtacct lnstat
+TARGETS=ss nstat
 
 include ../config.mk
 
-ifeq ($(HAVE_BERKELEY_DB),y)
-	TARGETS += arpd
-endif
+#ifeq ($(HAVE_BERKELEY_DB),y)
+#	TARGETS += arpd
+#endif
 
 all: $(TARGETS)
 
